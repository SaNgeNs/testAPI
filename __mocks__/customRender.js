import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import createStore from 'Store';

export const createWrapper = (initialState) => {
  const wrapperStore = createStore(initialState);

  const AllProviders = ({ children }) => (
    <Provider store={wrapperStore}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  );

  return AllProviders;
};

const customRender = (ui, options, initialState) => render(ui, { wrapper: createWrapper(initialState), ...options });

export default customRender;
