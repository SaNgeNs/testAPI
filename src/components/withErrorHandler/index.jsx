import React from 'react';

export function withErrorHandler(Component) {
  class WithErrorHandler extends React.Component {
    constructor() {
      super();

      this.state = {
        hasError: false,
      };
    }

    componentDidCatch(error, info) {
      console.error('error: ', error);
      console.error('errorInfo: ', info);

      this.setState({
        hasError: true,
      });
    }

    render() {
      const {
        hasError,
      } = this.state;

      if (hasError) {
        return null;
      }

      return <Component {...this.props} />;
    }
  }

  return WithErrorHandler;
}

export default withErrorHandler;
