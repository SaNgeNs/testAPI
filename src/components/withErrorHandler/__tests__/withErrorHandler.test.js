import React from 'react';
import customRender from 'Mock/customRender';
import { getNodeText } from '@testing-library/react';

import withErrorHandler from '../index';

describe('withErrorHandler: ', () => {
  it(`should return null`, () => {
    const Throws = () => {
      throw new Error('Oh no!')
    };

    const ErrorHandler = withErrorHandler(Throws);
    jest.spyOn(global.console, 'error').mockImplementation(() => jest.fn());

    const { container, unmount } = customRender(
      <ErrorHandler />
    );

    expect(getNodeText(container)).toBe('');
    expect(container).toMatchSnapshot();
    unmount();
  });
});
