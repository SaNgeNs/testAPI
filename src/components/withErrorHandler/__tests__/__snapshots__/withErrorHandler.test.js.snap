// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withErrorHandler:  should return null 1`] = `<div />`;
